src/MessageTrait.php

```diff
@@ -196,21 +152,39 @@ private function setHeaders(array $headers): void
     }
 
     /**
-     * Trims whitespace from the header values.
+     * Make sure header has a non-empty string name and a sting value.
      *
-     * Spaces and tabs ought to be excluded by parsers when extracting the field value from a header field.
+     * Trims whitespace from the header values. Spaces and tabs ought to be
+     * excluded by parsers when extracting the field value from a header field.
      *
      * header-field = field-name ":" OWS field-value OWS
      * OWS          = *( SP / HTAB )
      *
-     * @param string[] $values Header values
-     *
-     * @return string[] Trimmed header values
-     *
      * @see https://tools.ietf.org/html/rfc7230#section-3.2.4
      */
-    private function trimHeaderValues(array $values): array
+    private function validateAndTrimHeader($header, $values): array
     {
+        if (!is_array($values)) {
+            $values = [$values];
+        } elseif (empty($values)) {
+            throw new \InvalidArgumentException('Header values must be strings, empty array given.');
+        } else {
+            // Non empty array
+            $values = array_values($values);
+        }
+
+        if (!is_string($header) || empty($header)) {
```

Comment:
Should we also check that the header name is allowed per RFC 7230? More specifically this means it consists of 1 or more `tchar` exclusively:

```
tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
               / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
               / DIGIT / ALPHA
DIGIT          = %x30-39
ALPHA          = %x41-5A / %x61-7A
```

This will also make sure we never accidentally add a header name ending in whitespace, which is [explicitly forbidden](https://tools.ietf.org/html/rfc7230#section-3.2.4):

> No whitespace is allowed between the header field-name and colon. In the past, differences in the handling of such whitespace have led to security vulnerabilities in request routing and response handling.

Reply:
That would be great! Can you make that change?